File: LeetCode/Java/154-array-寻找旋转排序数组中的最小值-ii.java
/*
 * @lc app=leetcode.cn id=154 lang=java
 *
 * [154] 寻找旋转排序数组中的最小值 II
 */

// @lc code=start
class Solution {
    public int findMin(int[] nums) {
        for (int i = 1; i < nums.length; i++)
            if (nums[i] < nums[i - 1])
                return nums[i];
        return nums[0];
    }
}
// @lc code=end

File: LeetCode/Go/154-array-寻找旋转排序数组中的最小值-ii.go
/*
 * @lc app=leetcode.cn id=154 lang=golang
 *
 * [154] 寻找旋转排序数组中的最小值 II
 */
package leetcode

// @lc code=start
func findMin(nums []int) int {
	l, r := 0, len(nums)-1
	for l < r {
		p := l + (r-l)>>1
		if nums[p] < nums[r] {
			r = p
		} else if nums[p] > nums[r] {
			l = p + 1
		} else {
			r--
		}
	}
	return nums[l]
}

// @lc code=end
